### ISSUE TITLE
Postman not redirecting with headers, even with "Retain headers"

### ISSUE THREAD

**Issue by DavideCarvalho** [a85/POSTMan-Chrome-Extension#6013]
## App Details:
I can't, this is an internal app
```
Postman for Windows
Latest version - I don't know the number -
Windows 10 Enterprise / x64
```

## Issue Report:
1. I have a route that redirects to another api
2. Enabled "Retain headers when clicking on links"
3. Redirected api gave me "unauthorized"
4. Disabled "Automatically follow redirects"
5. When making the request, postman gives me the link to the other api. I click and the header is there
6. Enable again "Automatically follow redirects" and the unauthorized still persists

Create a route that redirects to another api and see if the header is being passed. If you disable "Automatically follow redirects", it shows the headers, but enabling it doesn't pass the headers forward

**Comment by rosponda57** [comment]
I hope it will be fixed soon...

**Comment by codenirvana** [comment]
@DavideCarvalho @rosponda57 We do retain request headers on redirects. The only exception is:
`Authorization` header will be dropped if the request redirects to a different host (e.g, abc.com -> xyz.com).

Is that the case? Can you provide more details about the redirect?

**Comment by rosponda57** [comment]
He,
Thanks for interest,

Im positive that its the same problem,
We are using a site with redirect,  the version 7.0.7 works good, the 7.0.8
seems not to send headers to the redirect link in Header:

[image: image.png]


In the 7.0.7 version , the redirect is followed (when "Automatically
follow  redirects  " option is on ) with no problem
On version 7.0.8 we get :
[image: image.png]


When on v. 7.0.8 we turn  "Automatically follow redirects" off and we copy
the redirect link to the address bar (we simulate 307 redirect action)
everything is ok:

[image: image.png]

if You need more info let me know

Best regards,
Robert





pt., 12 kwi 2019 o 20:39 Udit Vasu <notifications@github.com> napisał(a):

**Comment by rosponda57** [comment]
> @DavideCarvalho @rosponda57 We do retain request headers on redirects. The only exception is:
> `Authorization` header will be dropped if the request redirects to a different host (e.g, abc.com -> xyz.com).
> 
> Is that the case? Can you provide more details about the redirect?

OK, i'll check what kind of redirect (subdomain, entire domain) is it.

**Comment by rosponda57** [comment]
The problem is when we change subdomain  , when redirect from aaa.mydomail.com to bbb.mydomail.com the authorization header is cleaned, (even if the "Retain headers when clicking on links" is on, but i'm not sure if it's proper in this case) .

**Comment by rosponda57** [comment]
Version 7.0.9 has the same problem.
The Authorization header  is cleaned when redirect is made between subdomains

Edit:
I've downgreaded to 7.0.7 and redirect between subdomains works well

**Comment by codenirvana** [comment]
@rosponda57 This is expected behavior, `Authorization` header is not supposed to be retained when redirecting to a different host (yes, sub-domain as well i.e, abc.com -> xyz.abc.com).

In v7.0.8 we fixed this bug for 307 and 308 status codes.
Related curl vulnerability: https://curl.haxx.se/docs/CVE-2018-1000007.html

To fix this issue we will add a request configuration (similar to curl's UNRESTRICTED_AUTH) which will retain the `Authorization` header for other hosts too.
Will update you once it's added, follow the original thread #3703 for the same issue.

Closing as duplicate of #3703
